controlplane/kubeadm/controllers/upgrade.go

```diff
@@ -85,13 +85,7 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, errors.Wrap(err, "failed to upgrade kubelet config map")
 	}
 
-	status, err := workloadCluster.ClusterStatus(ctx)
-	if err != nil {
-		return ctrl.Result{}, err
-	}
-
-	if status.Nodes <= *kcp.Spec.Replicas {
-		// scaleUp ensures that we don't continue scaling up while waiting for Machines to have NodeRefs
+	if controlPlane.NeedsScaleUp() {
 		return r.scaleUpControlPlane(ctx, cluster, kcp, controlPlane)
 	}
 	return r.scaleDownControlPlane(ctx, cluster, kcp, controlPlane)
```

Comment:
Will we ever reach this now that there is a check for `!c.NeedsScaleDown()` in the `NeedsUpgrade()` check that gates this logic?

Reply:
Ah true yes this is now just a special scale-up